Sources/ComposableArchitecture/TestSupport/TestStore.swift

```diff
@@ -233,7 +233,9 @@
             .eraseToEffect { .init(origin: .receive($0), file: action.file, line: action.line) }
 
         },
-        environment: ()
+        environment: (),
+        mainThreadChecksEnabled: true,
```

Comment:
I've kept `true` as it was the current value, but it possible that we would want `false`. In this case, we could make `Store.init` private again and use `Store.unchecked` for `TestStore`.
I'm disabling RW for `TestStore` because of the failure I had in CI a few commits above. I didn't try to reproduce it, and it is also probable that disabling them is no needed after all.

Reply:
Maybe runtime warnings should be promoted into test failures, but I don't know if this is possible without refactoring too much. This is not the purpose of this PR anyway.